data/publisher/src/main/java/tech/pegasys/teku/data/publisher/PrometheusMetricsPublisherSource.java

```diff
@@ -47,14 +62,119 @@ public int getValidatorsActive() {
     return validatorsActive;
   }
 
+  @Override
+  public long getHeadSlot() {
+    return headSlot;
+  }
+
+  @Override
+  public boolean isValidatorPresent() {
+    return validatorsTotal > 0;
+  }
+
+  @Override
+  public boolean isBeaconNodePresent() {
+    return isBeaconNodePresent;
+  }
+
+  @Override
+  public boolean isEth2Synced() {
+    return isEth2Synced;
+  }
+
+  @Override
+  public boolean isEth1Connected() {
+    return isEth1Connected;
+  }
+
+  @Override
+  public int getPeerCount() {
+    return peerCount;
+  }
+
+  @Override
+  public long getGossipBytesTotalSent() {
+    return gossipBytesTotalSent;
+  }
+
+  @Override
+  public long getGossipBytesTotalReceived() {
+    return gossipBytesTotalReceived;
+  }
+
   private void storeObservationIfNeeded(final Observation observation) {
+    MetricCategory category = observation.getCategory();
+    if (category.equals(EXECUTOR)) {
+      readExecutorCategoryItem(observation);
+    } else if (category.equals(PROCESS)) {
+      readProcessCategoryItem(observation);
+    } else if (category.equals(VALIDATOR)) {
+      readValidatorCategoryItem(observation);
+    } else if (category.equals(JVM)) {
+      readJvmCategoryItem(observation);
+    } else if (category.equals(BEACON)) {
+      readBeaconCategoryItem(observation);
+    } else if (category.equals(LIBP2P)) {
+      readLibP2pCategoryItem(observation);
+    }
+  }
+
+  private void readLibP2pCategoryItem(final Observation observation) {
+    if (observation.getMetricName().equals("gossip_bytes_total")) {
+      if (observation.getLabels().contains("received_bytes")) {
+        gossipBytesTotalReceived = getLongValue(observation.getValue());
+      } else if (observation.getLabels().contains("sent_bytes")) {
+        gossipBytesTotalSent = getLongValue(observation.getValue());
+      }
+    }
+  }
+
+  private void readBeaconCategoryItem(final Observation observation) {
+    isBeaconNodePresent = true;
+    switch (observation.getMetricName()) {
+      case "head_slot":
+        headSlot = getLongValue(observation.getValue());
+        break;
+      case "eth1_request_queue_size":
+        isEth1Connected = true;
+        break;
+      case "peer_count":
+        peerCount = getIntValue(observation.getValue());
+        break;
+    }
+  }
+
+  private void readExecutorCategoryItem(final Observation observation) {
     switch (observation.getMetricName()) {
       case "cpu_seconds_total":
         cpuSecondsTotal = getLongValue(observation.getValue());
         break;
       case "memory_pool_bytes_used":
         addToMemoryPoolBytesUsed((Double) observation.getValue());
         break;
```

Comment:
Should these still be here under executor? Looks like we have them under process and jvm.

Reply:
probably i haven't cleaned them up, yep